### DISCUSSION TITLE
Some questions regarding to Unified Endpoint proposal

### DISCUSSION THREAD

**Discussion by hitanshu-mehta** [thanos-io/thanos#4003] [Questions & Answers] [answered]
Hi @lilic @bwplotka @brancz! I am planning to work on a unified endpoint [proposal](https://thanos.io/tip/proposals/202101_endpoint_discovery.md/). I have some questions regarding the proposal.
1. is Info API a new gRPC Service? A new RPC in an existing gRPC Service? Right now we are mixing terminology a lot and interchangeably using API and service in the alternative and the main proposal.
2. @brancz  listed the fields for the new endpoint as a [comment](https://github.com/thanos-io/thanos/pull/3703#discussion_r553971668) to an alternative solution but it is added in the main solution in the proposal.

Thank you :)

**Comment by bwplotka** [comment, marked as answer]
Thanks @hitanshu-mehta !

1. Very good question. I think for Info we talked about API as for whole new gRPC Service. Reason is that we already have Info method! (: 

2. Yes, it was changed from alternative to main solution. So I think his comments are already part of main solution now.

**Reply by hitanshu-mehta** [reply]
So, the main solution is proposing to add a new Info service? If yes, then how alternative solution is different from the main one, because the alternative solution is also talking about splitting the Info method out of the Store API into a separate gRPC service.

**Reply by bwplotka** [reply]
Let's be specifc. I have kind of lost what alternative you are talking about. Frederic comment you linked is part of main solution now

**Reply by hitanshu-mehta** [reply]
Sorry for being unclear 😅 . I'm talking about the alternative solution mentioned in the proposal itself. [https://github.com/thanos-io/thanos/blob/main/docs/proposals/202101_endpoint_discovery.md#alternatives](url)

**Reply by hitanshu-mehta** [reply]
Discussed in contributors' office hours. Yes, both proposals are similar.
